cmd/kthena-router/app/router.go

```diff
@@ -345,13 +377,23 @@ func (lm *ListenerManager) createPortHandler(port int32) gin.HandlerFunc {
 			return
 		}
 
-		// Handle /v1/*path
+		// Try to handle with ModelRoute first (for /v1/* paths)
 		if strings.HasPrefix(c.Request.URL.Path, "/v1/") {
-			lm.router.HandlerFunc()(c)
+			modelRoutes := lm.store.GetModelRoutesByGateway(listenerConfig.GatewayKey)
+			if len(modelRoutes) > 0 {
+				lm.router.HandlerFunc()(c)
+				return
+			}
+		}
+
+		// Try to handle with HTTPRoute
+		httpRoutes := lm.store.GetHTTPRoutesByGateway(listenerConfig.GatewayKey)
+		if len(httpRoutes) > 0 {
+			lm.handleHTTPRoute(listenerConfig.GatewayKey)(c)
```

Comment:
There is still a gap from what we discussed: currently if both modelRoute and HTTPRoute is bounded to a gateway. We will only handle request with modelRoute. But if no match satisfied, then we can incorrectly return a 404.